core/src/test/java/eu/fasten/core/data/RevisionCallGraphTest.java

```diff
@@ -1,209 +1,204 @@
+        d = new RevisionCallGraph.Dependency(json);
+        assertEquals("maven", d.forge);
+        assertEquals("foo.bar", d.product);
+        List<RevisionCallGraph.Constraint> constraints = d.constraints;
+        assertEquals(3, constraints.size());
+        assertEquals("3.1", constraints.get(0).lowerBound);
+        assertEquals("7.1", constraints.get(0).upperBound);
+        assertEquals("9", constraints.get(1).lowerBound);
+        assertEquals("9", constraints.get(1).upperBound);
+        assertEquals("10.3", constraints.get(2).lowerBound);
+        assertNull(constraints.get(2).upperBound);
 
         assertJsonEquals(new JSONObject(spec.replaceAll(" ", "")), d.toJSON());
-	}
-
-	@Test
-	public void testDepset() {
-		String spec = "[" +
-				"[{\"forge\": \"maven\", \"product\": \"foo.bar\", \"constraints\": [\"[3.1..  7.1   ]\",\"[   9]\",\"[10.3  ..]\"] }], " +
-				"[{\"forge\": \"other\", \"product\": \"bar.nee\", \"constraints\": [\"[..9]\",\"[10.3  ..]\"] }]" +
-				"]";
-		JSONArray json = new JSONArray(spec);
-		List<List<RevisionCallGraph.Dependency>> depset = RevisionCallGraph.Dependency.depset(json);
-		RevisionCallGraph.Dependency d = depset.get(0).get(0);
-		assertEquals("maven", d.forge);
-		assertEquals("foo.bar", d.product);
-		List<RevisionCallGraph.Constraint> constraints = d.constraints;
-		assertEquals(3, constraints.size());
-		assertEquals("3.1", constraints.get(0).lowerBound);
-		assertEquals("7.1", constraints.get(0).upperBound);
-		assertEquals("9", constraints.get(1).lowerBound);
-		assertEquals("9", constraints.get(1).upperBound);
-		assertEquals("10.3", constraints.get(2).lowerBound);
-		assertNull(constraints.get(2).upperBound);
-		d = depset.get(1).get(0);
-		assertEquals("other", d.forge);
-		assertEquals("bar.nee", d.product);
-		constraints = d.constraints;
-		assertEquals(2, constraints.size());
-		assertNull(constraints.get(0).lowerBound);
-		assertEquals("9", constraints.get(0).upperBound);
-		assertEquals("10.3", constraints.get(1).lowerBound);
-		assertNull(constraints.get(1).upperBound);
+    }
+
+    @Test
+    public void testDepset() {
+        String spec = "[" +
+                "[{\"forge\": \"maven\", \"product\": \"foo.bar\", \"constraints\": [\"[3.1..  7.1   ]\",\"[   9]\",\"[10.3  ..]\"] }], " +
+                "[{\"forge\": \"other\", \"product\": \"bar.nee\", \"constraints\": [\"[..9]\",\"[10.3  ..]\"] }]" +
+                "]";
+        JSONArray json = new JSONArray(spec);
+        List<List<RevisionCallGraph.Dependency>> depset = RevisionCallGraph.Dependency.depset(json);
+        RevisionCallGraph.Dependency d = depset.get(0).get(0);
+        assertEquals("maven", d.forge);
+        assertEquals("foo.bar", d.product);
+        List<RevisionCallGraph.Constraint> constraints = d.constraints;
+        assertEquals(3, constraints.size());
+        assertEquals("3.1", constraints.get(0).lowerBound);
+        assertEquals("7.1", constraints.get(0).upperBound);
```

Comment:
Does `Constraint` have hashCode and equals implemented? If yes, this type of test could be reduced to a single assert between `new Constraint(...)` and `constraints.get(0)`, avoiding two asserts for the two fields (the same is true for the following fields as well). There might be a chance to reduce the size of this test suite.

Reply:
Will be added when refactoring RCG